extensions/sql-database-projects/src/dialogs/deployDatabaseDialog.ts

```diff
@@ -115,42 +116,47 @@ export class DeployDatabaseDialog {
 	}
 
 	public async getConnectionUri(): Promise<string> {
-		// if target connection is a data source, have to check if already connected or if connection dialog needs to be opened
-		let connId: string;
-
-		if (this.dataSourcesRadioButton?.checked) {
-			const dataSource = (this.dataSourcesDropDown!.value! as DataSourceDropdownValue).dataSource;
-
-			const connProfile: azdata.IConnectionProfile = {
-				serverName: dataSource.server,
-				databaseName: dataSource.database,
-				connectionName: dataSource.name,
-				userName: dataSource.username,
-				password: dataSource.password,
-				authenticationType: dataSource.integratedSecurity ? 'Integrated' : 'SqlAuth',
-				savePassword: false,
-				providerName: 'MSSQL',
-				saveProfile: true,
-				id: dataSource.name + '-dataSource',
-				options: []
-			};
-
-			if (dataSource.integratedSecurity) {
-				connId = (await this.apiWrapper.connectionConnect(connProfile, false, false)).connectionId;
+		try {
+			// if target connection is a data source, have to check if already connected or if connection dialog needs to be opened
+			let connId: string;
+
+			if (this.connectionIsDataSource) {
```

Comment:
This line is the relevant part of the fix.  The rest is just whitespace changes due to the try-catch wrap.